### DISCUSSION TITLE
background coloring for pyqtgraph.dockarea

### DISCUSSION THREAD

**Discussion by armeya25** [highfestiva/finplot#481] [Q&A]
i am trying to set background color to DockArea, but no information vailable to it,
how do i do it.

**Comment by armeya25** [comment]
sorry i got it now,
fplt.foreground = '#D6DBDF'
fplt.background = '#151E26'

these must be added befor DockArea initialisation

still dont know why central dock doesnt get color





#!/usr/bin/env python3

import finplot as fplt
from PyQt6.QtWidgets import QApplication, QGridLayout, QMainWindow, QGraphicsView, QComboBox, QLabel
from pyqtgraph.dockarea import DockArea, Dock

from resample import Resample_DataFrame
import polars as pl
import pandas as pd

### mainly color updations for fplt
fplt.cross_hair_color = 'white'
fplt.foreground = '#D6DBDF'
fplt.background = '#151E26'
bull = '#1ABC9C'
bear = '#E74C3C'
fplt.candle_bull_color = bull
fplt.candle_bull_body_color = bull
fplt.candle_bear_color = bear

app = QApplication([])
win = QMainWindow()
area = DockArea()
win.setCentralWidget(area)
win.resize(1600,800)
win.setWindowTitle("Docking charts example for finplot")

# Set width/height of QSplitter
win.setStyleSheet("QSplitter { width : 20px; height : 10px; }")

# Create docks
ltf_dock = Dock("lower time", size = (1000, 100), closable = False)
itf_df = Dock("intermediate time", size = (1000, 30), closable = False)
htf_df = Dock("higher time", size = (1000, 40), closable = False)
area.addDock(ltf_dock)
area.addDock(itf_df)
area.addDock(htf_df)

# Chart for docks
ax0,ax1,ax2 = fplt.create_plot_widget(master=area, rows=3)
area.axs = [ax0, ax1, ax2]
ltf_dock.addWidget(ax0.ax_widget, 1, 0, 1, 2)
itf_df.addWidget(ax1.ax_widget, 1, 0, 1, 2)
htf_df.addWidget(ax2.ax_widget, 1, 0, 1, 2)

# Link x-axis
ax0.setXLink(ax2)
ax1.setXLink(ax2)
#ax1.setXLink(ax0)
#ax2.setXLink(ax0)
win.axs = [ax0,ax1,ax2]

## dataframes
df = pl.read_csv('data/nifty/nifty.csv')

d5m = Resample_DataFrame(df, '5m')
d5m = d5m.with_columns(pl.col('date').dt.epoch(time_unit="s").alias("date"))
ltf_df = d5m.to_pandas().set_index('date')

d15m = Resample_DataFrame(df, '15m')
d15m = d15m.with_columns(pl.col('date').dt.epoch(time_unit="s").alias("date"))
itf_df = d15m.to_pandas().set_index('date')

d75m = Resample_DataFrame(df, '1d')
d75m = d75m.with_columns(pl.col('date').dt.epoch(time_unit="s").alias("date"))
htf_df = d75m.to_pandas().set_index('date')

ltf_df = ltf_df["open close high low".split()]
itf_df = itf_df["open close high low".split()]
htf_df = htf_df["open close high low".split()]

plot_5min = fplt.candlestick_ochl(ltf_df, candle_width=0.8, ax = ax0)
plot_15min = fplt.candlestick_ochl(itf_df,candle_width=0.8, ax = ax1)
plot_75min = fplt.candlestick_ochl(htf_df, candle_width=0.8, ax = ax2)

## now update color for each dock chart
plot_5min.colors.update({
            'bull_body': fplt.candle_bull_color
        })
plot_15min.colors.update({
            'bull_body': fplt.candle_bull_color
        })
plot_75min.colors.update({
            'bull_body': fplt.candle_bull_color
        })

fplt.show(qt_exec = False) # prepares plots when they're all setup
win.show()
app.exec()

**Comment by armeya25** [comment]
![IMG_20231225_100857_699](https://github.com/highfestiva/finplot/assets/135110431/3af083a5-59a4-4970-b12b-7911804e2b7e)


This is what I am getting,
Although I want all dock background as black

**Comment by highfestiva** [comment]
There are two background colors, intended for alternating rows in order to distinguish them better. They are called `background` (for even rows, starting at 0) and `odd_plot_background` (for odd rows, starting at 1). Setting `odd_plot_background='#151E26'` is probably what you're looking for.

**Comment by armeya25** [comment]
Thank you
